src/main/java/dk/cs/aau/huppaal/HUPPAALPreloader.java

```diff
@@ -0,0 +1,56 @@
+package dk.cs.aau.huppaal;
+
+import dk.cs.aau.huppaal.presentations.PreloaderPresentation;
+import javafx.application.Platform;
+import javafx.application.Preloader;
+import javafx.scene.Scene;
+import javafx.stage.Stage;
+import javafx.stage.StageStyle;
+
+public class HUPPAALPreloader extends Preloader {
+    public enum LoadStage {
+        LOADING_PROJECT,
+        INITALIZE_JFX,
+        AFTER_INIT,
+        START_JFX,
+        AFTER_SHOW,
+        FINISHED
+    }
+    public static class Notification implements PreloaderNotification {
+        private final LoadStage stage;
+        public Notification(LoadStage stage) {
+            this.stage = stage;
+        }
+        public LoadStage getStage() {
+            return stage;
+        }
+    }
+    private Stage stage;
+    private PreloaderPresentation presentation;
+
+    @Override
+    public void start(Stage primaryStage) throws Exception {
+        stage = new Stage();
+        presentation = new PreloaderPresentation();
+        presentation.getController().projectNameLabel.setText(HUPPAAL.projectDirectory.getValue());
+        stage.setScene(new Scene(presentation));
+        stage.initStyle(StageStyle.UNDECORATED);
+        stage.show();
+    }
+
+    @Override
+    public void handleApplicationNotification(PreloaderNotification info) {
+        if(info instanceof Notification ninfo) {
+            System.out.println(ninfo.getStage().name());
```

Comment:
Should this be removed before merge?

Reply:
Removed in 39413a4ac07ad909811fc4b27db8f143d0f58ee2